utils.py

```diff
@@ -116,21 +114,20 @@ def get_exploration_components_from_dir(dir_path):
         for filename in files:
             filepath = os.path.join(root, filename)
             if root == dir_path:
-                if filepath.endswith('.DS_Store'):
```

Comment:
I changed the structure of the code as I don't know if it will be a good idea to create a .DS_Store file just to test this.

Reply:
I think this change is fine.